core/src/main/scala/org/apache/spark/storage/StorageLevel.scala

```diff
@@ -60,10 +60,7 @@ class StorageLevel private(
   assert(replication < 40, "Replication restricted to be less than 40 for calculating hash codes")
 
   if (useOffHeap) {
-    require(!useDisk, "Off-heap storage level does not support using disk")
-    require(!useMemory, "Off-heap storage level does not support using heap memory")
     require(!deserialized, "Off-heap storage level does not support deserialized storage")
```

Comment:
as discussed offline, we should also require that `offHeap` implies `useMemory`.

(Additionally `useDisk` should imply `!deserialized`, but we can do that later if you prefer)